packages/neon-core/src/tx/transaction/EnrollmentTransaction.ts

```diff
@@ -0,0 +1,83 @@
+import {
+  getPublicKeyEncoded,
+  getPublicKeyUnencoded
+} from "../../../src/wallet/core";
+import { TX_VERSION } from "../../consts";
+import { DEFAULT_SYSFEE } from "../../consts";
+import { num2hexstring, num2VarInt, reverseHex, StringStream } from "../../u";
+import { BaseTransaction, TransactionLike } from "./BaseTransaction";
+import TransactionType from "./TransactionType";
+
+export interface EnrollmentTransactionLike extends TransactionLike {
+  publicKey: string;
+}
+
+export class EnrollmentTransaction extends BaseTransaction {
+  public static deserializeExclusive(
+    ss: StringStream,
+    tx: Partial<TransactionLike>
+  ): Partial<EnrollmentTransactionLike> {
+    const hexStrPrefix = ss.read(1);
+    const prefix = parseInt(hexStrPrefix, 16);
+
+    let pKey: string = "";
+    // Compressed public keys.
+    if (prefix === 0x02 || prefix === 0x03) {
+      pKey = ss.read(32);
+    } else if (prefix === 0x04) {
+      pKey = ss.read(65);
```

Comment:
wouldnt this be `read(64)`? Since we have already read the first bit off in all cases.